### PR TITLE
Writing StoreFindToken and BlobReadOptions in new formats

### PR DESCRIPTION
**Pull request by vgkholla**
This commit writes `StoreFindToken` and `BlobReadOptions` in the new formats and thus enables log segmentation end to end

Moved `StoreFindToken` into its own file to make it `public` so that it can be used in the server integration tests.

`./gradlew build && ./gradlew test` passed successfully on both OSx and Linux.

**Primary reviewers : @pnarayanan, @nsivabalan**
**Expected time to review : 15 mins**

### CODE REVIEW

File: ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
+/**
```
Comment (vgkholla) [inline ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java]: Actual changes in this file are the capitalization of the static constants and the implementation of `toBytes()`.

**Comment by nsivabalan** [comment]
Looks good to me.

File: ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+
+
+/**
+ * The StoreFindToken is an implementation of FindToken.
+ * It is used to provide a token to the client to resume
+ * the find from where it was left previously. The StoreFindToken
+ * maintains a offset to track entries within the journal. If the
```
Comment (pnarayanan) [inline ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java]: an offset

File: ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    Uninitialized, JournalBased, IndexBased
+  }
+
+  private Type type;
+  private Offset offset;
+  private StoreKey storeKey;
+  private UUID sessionId;
+  private long bytesRead;
+
+  private static final short VERSION = 1;
```
Comment (pnarayanan) [inline ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java]: I think this should be named as `VERSION_1` and you should use another string constant (`VERSION_0`) for the previous version.

File: ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+  private static final int VERSION_SIZE = 2;
+  private static final int TYPE_SIZE = 2;
+  private static final int SESSION_ID_SIZE = 4;
```
Comment (pnarayanan) [inline ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java]: this is actually `SESSION_ID_LENGTH_SIZE`, isn't it? And 2 bytes should suffice really (if you wanted to make that change, now would be a good time).

File: ambry-store/src/main/java/com.github.ambry.store/StoreMessageReadSet.java [hunk 5]
```diff
@@ -41,9 +41,8 @@
   private final AtomicBoolean open = new AtomicBoolean(true);
   private final Logger logger = LoggerFactory.getLogger(getClass());
 
-  private static final short VERSION = 0;
+  private static final short VERSION = 1;
```
Comment (pnarayanan) [inline ambry-store/src/main/java/com.github.ambry.store/StoreMessageReadSet.java]: perhaps you could have `VERSION_0` and `VERSION_1` defined for this class as well.

File: ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java (hunk 2 above)
Comment (vgkholla) [inline ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java]: Fixed

File: ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java (hunk 3 above)
Comment (vgkholla) [inline ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java]: done. Also added tests for ser/deser in both versions.

File: ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java (hunk 4 above)
Comment (vgkholla) [inline ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java]: name changed. Size remains the same.

**Comment by vgkholla** [comment]
@pnarayanan addressed comments. Formatted and build + test passed on OSX and Linux

**Comment by pnarayanan** [comment]
LGTM, merging.

**Pull request merged**